api/maven-api-core/src/main/java/org/apache/maven/api/services/RealmService.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.api.services;
+
+import org.apache.maven.api.Service;
+import org.apache.maven.api.annotations.Experimental;
+import org.apache.maven.api.annotations.Nonnull;
+import org.codehaus.plexus.classworlds.ClassWorld;
+import org.codehaus.plexus.classworlds.realm.ClassRealm;
+
+/**
+ * Access to {@link ClassRealm}.
+ *
+ * @since 4.0.0
+ */
+@Experimental
+public interface RealmService extends Service {
```

Comment:
not sure the goal but plexus should probably not surface in the API so maybe just use `ClassLoader`?

Reply:
Hm, you are right. Ultimate goal is "contain" scattered use of PlexusContainer. So let me change this PR, and offer single method w/ Classloader as you propose, but add "internal service" (usable in maven core only) that delivers what this PR currently has (as this is needed for 2 components that mis-used PlexusContainer only to obtain what they needed)